src/metatrain/utils/omegaconf.py

```diff
@@ -126,6 +126,86 @@ def _resolve_single_str(config: str) -> DictConfig:
 CONF_ENERGY["stress"] = CONF_GRADIENT.copy()
 
 
+def check_missing_options(target: DictConfig, reference: DictConfig):
+    """Validate that all keys in the ``target`` are present in ``reference``.
+
+    :param target: Target configuration object to be validated.
+    :param reference: Reference configuration object to validate against.
+
+    :raises ValueError: If a key in the target configuration is not found in the
+        reference configuration.
+    """
+    for key in target:
+        if key not in reference.keys():
+            raise ValueError(f"Config element {key!r} is an unknown element.")
+        elif isinstance(target[key], DictConfig) and isinstance(
+            reference[key], DictConfig
+        ):
+            check_missing_options(target[key], reference[key])
```

Comment:
After a bit of thinking, I think this function is the best we can do and I think it's also quite portable. Just one question: how do these `DictConfig`s work? I see that we're checking for DictConfigs recursively, but are all dicts in a DictConfing automatically also converted to `DictConfig`s?

Reply:
Yes everything which is either a `dictionary` or a `list` will be converted:

```python
from omegaconf import OmegaConf

d = {"a": {"inner_a": 5}, "b": [{"l_1":1}, {"l_2":2}, {"l_3":3}]}
conf = OmegaConf.create(d)

print("type a:", type(conf["a"]))
print("type b:", type(conf["b"]))
```

```
type a: <class 'omegaconf.dictconfig.DictConfig'>
type b: <class 'omegaconf.listconfig.ListConfig'>
```

Which means we can also add another branch if `target[key]` is a `ListConfig` and recursively call the function for all entries in the list.

Reply:
Perfect, thank you!